modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -1136,24 +1137,34 @@ class DeConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
 
         if (backendId == DNN_BACKEND_INFERENCE_ENGINE)
         {
-            if (kernel_size.size() == 3)
-                CV_Error(Error::StsNotImplemented, "Unsupported deconvolution3D layer");
+            if (kernel_size.size() == 3 && preferableTarget != DNN_TARGET_CPU) {
+                return false;
+            }
 
-            if (adjustPad.height || adjustPad.width)
+            if (std::accumulate(adjust_pads.begin(), adjust_pads.end(), 1, std::plus<size_t>()) > 0)
             {
                 if (padMode.empty())
                 {
-                    if (preferableTarget != DNN_TARGET_CPU && group != 1)
+                    if ((kernel_size.size() == 3 || preferableTarget != DNN_TARGET_CPU) && group != 1)
```

Comment:
Why we need to check `kernel_size.size() == 3` only?

Reply:
done